sql-plugin/src/main/scala/com/nvidia/spark/rapids/ColumnarRdd.scala

```diff
@@ -39,11 +38,16 @@ import org.apache.spark.sql.rapids.execution.InternalColumnarRddConverter
  * when they no longer need it.
  */
 object ColumnarRdd {
+
+  lazy val internalConverter: Class[_] = ShimLoader.newColumnarRDDClass()
+
   def apply(df: DataFrame): RDD[Table] = {
-    InternalColumnarRddConverter(df)
+    convert(df)
   }
 
   def convert(df: DataFrame): RDD[Table] = {
-    InternalColumnarRddConverter.convert(df)
+    internalConverter
+      .getDeclaredMethod("convert", classOf[DataFrame])
```

Comment:
Shouldn't we be caching the method rather than looking it up each time?

Reply:
yes updated